src/util/estimateNbPureComponents.js

```diff
@@ -0,0 +1,35 @@
+import { PCA } from 'ml-pca';
+
+/**
+ * Estimate the number of pure components of each range by NIPALS PCA
```

Comment:
Can you check if this method appears in the jsdoc ? It should not.

Maybe we could / shoud add @private ? https://jsdoc.app/tags-private.html